app/javascript/packs/column_form_modal.jsx

```diff
@@ -0,0 +1,128 @@
+import React from "react";
+import ReactDOM from "react-dom";
+import {
+  Button,
+  Modal,
+  ModalHeader,
+  ModalBody,
+  ModalFooter,
+  Form,
+  FormGroup,
+  Label,
+  Input
+} from "reactstrap";
+import axios from "axios";
+
+class ColumnFormModal extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      modal: false,
+      data: {
+        name: "",
+        project_id: this.props.project.id,
+      },
+      errors: {}
+    };
+    this.toggle = this.toggle.bind(this);
+  }
+  toggle() {
+    this.setState(prevState => ({
+      modal: !prevState.modal
+    }));
+  }
+  setFromValue = (attribute, value) => {
+    this.setState(prev => ({ data: { ...prev.data, [attribute]: value } }));
+  };
+  handleSubmit = e => {
+    if (this.handleValidation()) {
+      axios
+        .post(
+          `/projects/${this.state.data.project_id}/columns`,
+          { column: this.state.data },
+          {
+            headers: {
+              "X-CSRF-Token": document.querySelector('meta[name="csrf-token"]')
+                .content
+            }
+          }
+        )
+    }
+  };
+  handleValidation() {
+    let column = this.state.data;
+    let errors = {};
+    let formIsValid = true;
+    //title
+    if (!column.name) {
+      formIsValid = false;
+      errors["name"] = "Cannot be empty";
+    } else {
+      if (column.name.length > 30) {
+        formIsValid = false;
+        errors.name = "Must have less than 30 characters";
+      }
+    }
+
+    this.setState({ errors: errors });
+    return formIsValid;
+  }
+  render() {
+    return (
+      <>
+        <Button onClick={this.toggle} className="btn btn-primary">
+          +Add column
+        </Button>
+        <Modal
+          isOpen={this.state.modal}
+          toggle={this.toggle}
+          className={this.props.className}
+        >
+          <ModalHeader toggle={this.toggle} className="text-muted">
+            Add Column
+          </ModalHeader>
+          <Form onSubmit={this.handleSubmit}>
+            <ModalBody>
+              <FormGroup>
+                <Label for="name" className="text-muted">
+                  Name
+                </Label>
+                <Input
+                  type="text"
+                  name="title"
+                  id="title"
+                  placeholder="Title"
+                  value={this.state.data.title}
```

Comment:
this should be `this.state.data.name`, because you change `name` in your onChange function